interfaces/builtin/content_test.go

```diff
@@ -320,35 +320,209 @@ slots:
+  # inside.  This lets us perform the undo plan in case the writable
+  # mimic fails.
+  umount /snap/,
+  umount /snap/*,
+  umount /snap/*/,
+  # .. variant with mimic at /snap/consumer/
+  # Allow reading the mimic directory, it must exist in the first place.
+  /snap/consumer/ r,
+  # Allow setting the read-only directory aside via a bind mount.
+  /tmp/.snap/snap/consumer/ rw,
+  mount options=(rbind, rw) /snap/consumer/ -> /tmp/.snap/snap/consumer/,
+  # Allow mounting tmpfs over the read-only directory.
+  mount fstype=tmpfs options=(rw) tmpfs -> /snap/consumer/,
+  # Allow creating empty files and directories for bind mounting things
+  # to reconstruct the now-writable parent directory.
+  /tmp/.snap/snap/consumer/*/ rw,
+  /snap/consumer/*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/snap/consumer/*/ -> /snap/consumer/*/,
+  /tmp/.snap/snap/consumer/* rw,
+  /snap/consumer/* rw,
+  mount options=(bind, rw) /tmp/.snap/snap/consumer/* -> /snap/consumer/*,
+  # Allow unmounting the auxiliary directory.
+  # TODO: use fstype=tmpfs here for more strictness (LP: #1613403)
+  umount /tmp/.snap/snap/consumer/,
+  # Allow unmounting the destination directory as well as anything
+  # inside.  This lets us perform the undo plan in case the writable
+  # mimic fails.
+  umount /snap/consumer/,
+  umount /snap/consumer/*,
+  umount /snap/consumer/*/,
+  # .. variant with mimic at /snap/consumer/7/
+  # Allow reading the mimic directory, it must exist in the first place.
+  /snap/consumer/7/ r,
+  # Allow setting the read-only directory aside via a bind mount.
+  /tmp/.snap/snap/consumer/7/ rw,
   mount options=(rbind, rw) /snap/consumer/7/ -> /tmp/.snap/snap/consumer/7/,
+  # Allow mounting tmpfs over the read-only directory.
   mount fstype=tmpfs options=(rw) tmpfs -> /snap/consumer/7/,
-  mount options=(rbind, rw) /tmp/.snap/snap/consumer/7/** -> /snap/consumer/7/**,
+  # Allow creating empty files and directories for bind mounting things
+  # to reconstruct the now-writable parent directory.
+  /tmp/.snap/snap/consumer/7/*/ rw,
+  /snap/consumer/7/*/ rw,
+  mount options=(rbind, rw) /tmp/.snap/snap/consumer/7/*/ -> /snap/consumer/7/*/,
+  /tmp/.snap/snap/consumer/7/* rw,
+  /snap/consumer/7/* rw,
   mount options=(bind, rw) /tmp/.snap/snap/consumer/7/* -> /snap/consumer/7/*,
+  # Allow unmounting the auxiliary directory.
+  # TODO: use fstype=tmpfs here for more strictness (LP: #1613403)
   umount /tmp/.snap/snap/consumer/7/,
-  umount /snap/consumer/7{,/**},
-  /snap/consumer/7/** rw,
-  /snap/consumer/7/ rw,
-  /snap/consumer/ rw,
-  /tmp/.snap/snap/consumer/7/** rw,
-  /tmp/.snap/snap/consumer/7/ rw,
-  /tmp/.snap/snap/consumer/ rw,
-  /tmp/.snap/snap/ rw,
-  /tmp/.snap/ rw,
+  # Allow unmounting the destination directory as well as anything
+  # inside.  This lets us perform the undo plan in case the writable
+  # mimic fails.
+  umount /snap/consumer/7/,
+  umount /snap/consumer/7/*,
+  umount /snap/consumer/7/*/,
 `
```

Comment:
The `apparmor.WritableProfile(&buf, source, 1)` and `apparmor.WritableProfile(&buf, target, 1)` results in some somewhat unexpected rules (eg, `mount fstype=tmpfs options=(rw) tmpfs -> /,`) and quite a few duplicated rules. That's ok, apparmor will minimize the profile for stuff like this, but it will be nice when you implement the TODO. This is quite an improvement over before though, so not a blocker.